.github/workflows/build-src.yml

```diff
@@ -80,6 +80,7 @@ jobs:
         shell: bash
 
       - name: Restore ccache cache
+        if: inputs.build-target != 'linux64_fuzz'
```

Comment:
why does the fuzz build disable ccache?

Reply:
Good question. I have no idea tbh, I thought it was maybe conflicting somehow or wasn't useful. It was like that since it was originally introduced in #4732. @kwvg any hints?

Reply:
[Dropping `--disable-clang`](https://github.com/UdjinM6/dash/commit/f30913fa12a76d44746134ce24d6b8b67f0a3c7c) and running it twice - [100% hit rate](https://github.com/UdjinM6/dash/actions/runs/20174554663/job/57921378320#step:9:2248), don't see any issues (though we don't really test CI fuzz builds in any way).

Reply:
@UdjinM6 no objections to enabling ccache, it is enabled upstream ([source](https://github.com/bitcoin/bitcoin/blob/ad8b23e589096776332ddae11878be03297ec04c/ci/test/00_setup_env_native_fuzz.sh#L18-L19)).

it might have been due to local issues (i remember `ccache` being a bit finicky back when i used it so it's likely i had disabled it when working on the PR and then kept it in the final iteration as it was known-good) but since CI is happy, no reason not to use it.